distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text.enso

```diff
@@ -69,6 +70,56 @@ type Text
             comparison_result = Text_Utils.compare_normalized self that
             Ordering.from_sign comparison_result
 
+    ## Checks whether `self` is equal to `that`, ignoring the case of the texts.
+
+       Arguments:
+       - that: The text to compare `self` for case-insensitive equality with.
+
+       Two texts are considered equal ignoring case if they are of the same length
+       and corresponding characters are equal ignoring case.
+
+       ! Unicode Equality
+         The definition of equality includes Unicode canonicalization. I.e. two
+         texts are equal if they are identical after canonical decomposition. This
+         ensures that different ways of expressing the same character in the
+         underlying binary representation are considered equal.
+
+       > Example
+         The string 'É' (i.e. the character U+00C9, LATIN CAPITAL LETTER E WITH
+         ACUTE) is equal ignore case to the string 'é' (i.e. the character U+00E9,
+         LATIN SMALL LETTER E WITH ACUTE), which is canonically the same as the
+         string 'e\u0301' (i.e. the letter `e` followed by U+0301, COMBINING ACUTE
+         ACCENT). Therefore:
+
+             (('É' . equals_ignore_case 'é') && ('é' . equals_ignore_case 'e\u0301')) == True
+    equals_ignore_case : Text -> Locale -> Boolean
```

Comment:
What rule are we going for to decide which methods on text should be here and which should stay in extensions?

Reply:
Just put the comparison and equality stuff in - wasn't planning on moving anything else in.